### PR TITLE
fix: Update repology multi package project filtering

### LINKED ISSUE #13365: Repology datasource package filtering not working as expected
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.15.0

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

github.com

### Describe the bug

When I add repology as datasource trough the regex manager, but renovate gives invalid version upgrade suggestions.

I have the following lines in my Dockerfile, and the regex manager and renovate github action configured
 ([Minimal example repository](url))

```Dockerfile
# renovate: datasource=repology depName=python3.8 lookupName=ubuntu_20_04/python3.8 versioning=loose
ARG V_PYTHON38="3.8.10-0ubuntu1~20.04.1"
# renovate: datasource=repology depName=wget lookupName=ubuntu_20_04/wget versioning=loose
ARG V_WGET="1.20.3-1ubuntu1"
```

## [What I get (at time of writing)](https://github.com/rv2673/renovate-test/pull/2):

 Package | Update | Change |
|---|---|---|
| python3.8 | minor | `3.8.10-0ubuntu1~20.04.1` -> `3.9.5-3ubuntu0~20.04.1` |
| wget | minor | `1.20.3-1ubuntu1` -> `1.99.1-2.1` |

Both of which don't exist for the given packages.

## What I expect however is the following:

| Package | Update | Change |
|---|---|---|
| python3.8 | minor | `3.8.10-0ubuntu1~20.04.1` -> `3.8.10-0ubuntu1~20.04.2` |
| wget | minor | `1.20.3-1ubuntu1` -> `1.20.3-1ubuntu2` |

## Cause

The versions for `python3.9` package and `wget2` package are returned as valid versions for the `python3.8` and `wget` packages respectively.

The reason that this happens because both the `python` and the `wget` repology projects contain multiple packages. 
The repology datasource code has a section for [this case](https://github.com/rv2673/renovate/blob/b34b099899cd6023406956d60f43f4e7303e23d4/lib/datasource/repology/index.ts#L98), but the filtering does not work as expected/described in the comment above it. The packages that don't have pkgType to be checked are not filtered out.

### Relevant debug logs

_No response_

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by rv2673**
This should fix https://github.com/renovatebot/renovate/issues/13365

## Changes:

Change the filtering of packages on package type and name if filtering on repo does not result in single match to work as described in the comment that was above that section.

Added tests to verify that the change results in the expected behaviour.

## Context:
https://github.com/renovatebot/renovate/issues/13365

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/repology/index.spec.ts [hunk 1]
```diff
@@ -343,5 +344,113 @@ describe('datasource/repology/index', () => {
...
+      const res = await getPkgReleases({
+        datasource,
+        versioning,
+        depName: 'some_repo/some-package',
+      });
+      expect(res).toMatchSnapshot();
+      expect(res.releases).toHaveLength(2);
+      expect(res.releases[0].version).toBe('1.0.0');
+      expect(res.releases[1].version).toBe('4.0.0');
+      expect(httpMock.getTrace()).toMatchSnapshot();
```
Comment (viceice) [inline lib/datasource/repology/index.spec.ts:430]:
```suggestion
```
Remove, obsolete

File: lib/datasource/repology/index.spec.ts [hunk 2]
```diff
@@ -343,5 +344,113 @@ describe('datasource/repology/index', () => {
...
+        status: 200,
+        body: pkgsJSON,
+      });
+
+      const res = await getPkgReleases({
+        datasource,
+        versioning,
+        depName: 'some_repo/some-package',
+      });
+      expect(res).toMatchSnapshot();
```
Comment (viceice) [inline lib/datasource/repology/index.spec.ts:426]: As res is small, make an inline `toEqual`, so it only needs one expect

File: lib/datasource/repology/index.spec.ts [hunk 3]
```diff
@@ -343,5 +344,113 @@ describe('datasource/repology/index', () => {
...
+      const res = await getPkgReleases({
+        datasource,
+        versioning,
+        depName: 'ubuntu_20_04/python3.8',
+      });
+      expect(res).toMatchSnapshot();
+      expect(res.releases).toHaveLength(2);
+      expect(res.releases[0].version).toBe('3.8.2-1ubuntu1');
+      expect(res.releases[1].version).toBe('3.8.10-0ubuntu1~20.04.2');
+      expect(httpMock.getTrace()).toMatchSnapshot();
```
Comment (viceice) [inline lib/datasource/repology/index.spec.ts:453]:
```suggestion
```
Remove

File: lib/datasource/repology/index.spec.ts [hunk 4]
```diff
@@ -343,5 +344,113 @@ describe('datasource/repology/index', () => {
...
+        status: 200,
+        body: fixturePython,
+      });
+
+      const res = await getPkgReleases({
+        datasource,
+        versioning,
+        depName: 'ubuntu_20_04/python3.8',
+      });
+      expect(res).toMatchSnapshot();
```
Comment (viceice) [inline lib/datasource/repology/index.spec.ts:449]: See comments above.

File: lib/datasource/repology/index.ts [hunk 5]
```diff
@@ -92,15 +92,14 @@ function findPackageInResponse(
...
   // We need to do additional filtering by matching allowed package types passed as params with package description.
   // Remaining packages are the one we are looking for
-  let packagesWithType: RepologyPackage[];
-  for (const pkgType of types) {
-    packagesWithType = repoPackages.filter(
-      (pkg) => !pkg[pkgType] || pkg[pkgType] === pkgName
-    );
-    if (packagesWithType.length === 1) {
-      break;
+  const packagesWithType = repoPackages.filter(function (pkg) {
```
Comment (viceice) [inline lib/datasource/repology/index.ts:95]:
```suggestion
  const packagesWithType = repoPackages.filter((pkg) => {
```
Use lambda function

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/datasource/repology/index.ts (hunk 5 above)
Comment (rv2673) [reply lib/datasource/repology/index.ts]: Done

File: lib/datasource/repology/index.spec.ts (hunk 3 above)
Comment (rv2673) [reply lib/datasource/repology/index.spec.ts]: Done

File: lib/datasource/repology/index.spec.ts (hunk 1 above)
Comment (rv2673) [reply lib/datasource/repology/index.spec.ts]: Done

File: lib/datasource/repology/index.spec.ts (hunk 2 above)
Comment (rv2673) [reply lib/datasource/repology/index.spec.ts]: Done

File: lib/datasource/repology/index.spec.ts (hunk 4 above)
Comment (rv2673) [reply lib/datasource/repology/index.spec.ts]: Done

**Review by viceice — APPROVED** [review]

**Pull request merged**